muskathlon/models/survey_user_input.py

```diff
@@ -31,12 +31,20 @@ def write(self, vals):
                     'muskathlon.stage_fill_profile').id)
             ])
 
-            # if conditions to check that the task is the correct one,
-            # same as survey in event definition
-            registrations.write({
-                'completed_task_ids': [
-                    (4, self.env.ref(
-                        'muskathlon.task_medical').id)
-                ]
-            })
+            surveys = registrations.mapped('event_id.medical_survey_id')
+            for survey in surveys:
```

Comment:
I would rather loop over the registrations and take one survey at a time. Because here you will write the last found survey in all registrations.
```
            for registration in registrations:
                survey = registration.event_id.medical_survey_id
                ...
                registration.write({'completed_task_ids': ...})
```